src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -129,7 +129,18 @@ public ResettableSqlStatement GetResettableStatement(string query)
             return new ResettableSqlStatement(statement);
         }
 
-        public void RunInTransaction(Action action)
+        public void RunInTransaction<TState>(Action<TState> action, TState state)
+        {
+            RunInTransaction(
+                state =>
+                {
+                    state.action(state.state);
+                    return (object)null;
```

Comment:
Maybe? 
```suggestion
                    return default(object);
```

Reply:
When I see `default(object)`, my first question is "why isn't this just `null`"? When I see `(object)null`, my first thought is "the cast is required for the type system".